### PR TITLE
Fix twig_to_array is deprecated

### PR DESCRIPTION
**Pull request by WebMamba**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| Issues        | 
| License       | MIT

### CODE REVIEW

**Comment by smnandre** [comment]
It's the same bug for all the pacakges failing in the CI ?

**Comment by WebMamba** [comment]
With this PR in complement #1330 the CI should be green again! 😁

File: src/TwigComponent/src/Twig/ComponentNode.php [hunk 1]
```diff
@@ -43,6 +43,16 @@ public function compile(Compiler $compiler): void
     {
         $compiler->addDebugInfo($this);
 
+        // since twig/twig 3.9.0: Using the internal "twig_to_array" function is deprecated.
+        $compiler
+            ->write('$toArray = function ($data) {')
+            ->write('if (method_exists(Twig\Extension\CoreExtension::class, \'toArray\')) {')
```
Comment (smnandre) [inline src/TwigComponent/src/Twig/ComponentNode.php]: I think this test should be done on compile time and not on render time... Or we will call method_exists a billion time per render..

So i'd make something like 

if (method_exists(...))  {
    $compiler->write('return Twig\Extension\CoreExtension::toArray($data);')
} else {
    $compiler->write('return twig_to_array'...)
}

Comment (weaverryan) [reply src/TwigComponent/src/Twig/ComponentNode.php]: That's a nice idea 👍

**Review by evertharmeling — APPROVED** [review]

**Comment by smnandre** [comment]
Could we merge this one @weaverryan @kbond  ?  

( if it's all good for you @WebMamba )

I'd like to bring the green back on the CI, having a better vision of current PR checks

**Comment by kbond** [comment]
Thanks Matheo.

**Pull request merged**